### ISSUE TITLE
[Bug]: VoiceOver announcing dropdown for Timepicker as "button"

### ISSUE THREAD

**Issue by UniqueClone** [microsoft/fluentui#27817]
### Library

React / v8 (@fluentui/react)

### System Info

```shell
System:
    OS: macOS 13.3.1
    CPU: (8) x64 Apple M1
    Memory: 38.69 MB / 8.00 GB
    Shell: 5.9 - /bin/zsh
Browsers:
    Firefox: 105.0.1
    Safari: 16.4
```

### Are you reporting Accessibility issue?

yes

### Reproduction

https://developer.microsoft.com/en-us/fluentui#/controls/web/timepicker

## Actual Behavior
When focus moves to button on TimePicker in "TimePicker with DatePicker", the button is announced as "button" with no context of labels or description.

## Expected Behavior
When focus moves to button on TimePicker in "TimePicker with DatePicker", the button should be announced with context or a description of what the button is/does.

I'm willing to help fix this issue if member of team will briefly assist in ramping up on codebase.

### Requested priority

Normal

### Are you willing to submit a PR to fix?

yes

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by kapilkashyap** [comment]
I am not sure if I am missing anything here but for some reason `yarn change` did not generate a `change` file.

Please suggest!
